From b1a147705e46abf76d233918e1d4502b8b3d1c88 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Fri, 9 Sep 2022 20:52:52 +0800
Subject: [PATCH] [riscv] Port [wasm][liftoff] Fix and cleanup tracing of
 return value

Port commit 6f9e71fa74eb589a48c0f5065ac961a64cb515a3

Change-Id: Id5226e0892f67573cea289040c2d5aa85f159478
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3886478
Commit-Queue: ji qiu <qiuji@iscas.ac.cn>
Reviewed-by: ji qiu <qiuji@iscas.ac.cn>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#83122}
---
 src/wasm/baseline/riscv/liftoff-assembler-riscv.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/wasm/baseline/riscv/liftoff-assembler-riscv.h b/src/wasm/baseline/riscv/liftoff-assembler-riscv.h
index 925b02a4624..fb2dcf62cc4 100644
--- a/src/wasm/baseline/riscv/liftoff-assembler-riscv.h
+++ b/src/wasm/baseline/riscv/liftoff-assembler-riscv.h
@@ -157,6 +157,10 @@ void LiftoffAssembler::PatchPrepareStackFrame(
   GenPCRelativeJump(kScratchReg, imm32);
 }
 
+void LiftoffAssembler::LoadSpillAddress(Register dst, int offset) {
+  SubWord(dst, fp, offset);
+}
+
 void LiftoffAssembler::FinishCode() { ForceConstantPoolEmissionWithoutJump(); }
 
 void LiftoffAssembler::AbortCompilation() { AbortedCodeGeneration(); }
-- 
2.35.1

